### ISSUE TITLE
[BUG] Scrolling error in logs: Invalid version format (non-numeric data) at /usr/bin/ddclient line 24.

### ISSUE THREAD

**Issue by josco95** [linuxserver/docker-ddclient#77]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Seeing the following scrolling in logs across two separate systems (Fedora Server 38 using podmain and QNAP QTS 5.1.1 using docker:

Oct 21 18:42:18 localhost ddclient[335021]: Invalid version format (non-numeric data) at /usr/bin/ddclient line 24.
Oct 21 18:42:19 localhost ddclient[335021]: Invalid version format (non-numeric data) at /usr/bin/ddclient line 24.
Oct 21 18:42:20 localhost ddclient[335021]: Invalid version format (non-numeric data) at /usr/bin/ddclient line 24.
Oct 21 18:42:21 localhost ddclient[335021]: Invalid version format (non-numeric data) at /usr/bin/ddclient line 24.
Oct 21 18:42:22 localhost ddclient[335021]: Invalid version format (non-numeric data) at /usr/bin/ddclient line 24.
Oct 21 18:42:23 localhost ddclient[335021]: Invalid version format (non-numeric data) at /usr/bin/ddclient line 24.
Oct 21 18:42:24 localhost ddclient[335021]: Invalid version format (non-numeric data) at /usr/bin/ddclient line 24.
Oct 21 18:42:26 localhost ddclient[335021]: Invalid version format (non-numeric data) at /usr/bin/ddclient line 24.
Oct 21 18:42:27 localhost ddclient[335021]: Invalid version format (non-numeric data) at /usr/bin/ddclient line 24.
Oct 21 18:42:28 localhost ddclient[335021]: Invalid version format (non-numeric data) at /usr/bin/ddclient line 24.
Oct 21 18:42:29 localhost ddclient[335021]: Invalid version format (non-numeric data) at /usr/bin/ddclient line 24.
Oct 21 18:42:30 localhost ddclient[335021]: Invalid version format (non-numeric data) at /usr/bin/ddclient line 24.

### Expected Behavior

Shouldn't scroll the error in logs

### Steps To Reproduce

Run the container. 

### Environment

```markdown
- OS:
- How docker service was installed:
```

### CPU architecture

x86-64

### Docker creation

```bash
docker run -d \
  --name=ddclient \
  -v $CONFIGS/config:/config \
  -e PGID=911 -e PUID=911  \
  -e TZ=America/New_York \
  linuxserver/ddclient
```

### Container logs

```bash
[migrations] started
[migrations] no migrations found
usermod: no changes
───────────────────────────────────────

      ██╗     ███████╗██╗ ██████╗
      ██║     ██╔════╝██║██╔═══██╗
      ██║     ███████╗██║██║   ██║
      ██║     ╚════██║██║██║   ██║
      ███████╗███████║██║╚██████╔╝
      ╚══════╝╚══════╝╚═╝ ╚═════╝

   Brought to you by linuxserver.io
───────────────────────────────────────

To support LSIO projects visit:
https://www.linuxserver.io/donate/

───────────────────────────────────────
GID/UID
───────────────────────────────────────

User UID:    911
User GID:    911
───────────────────────────────────────

[custom-init] No custom files found, skipping...
Setting up watches.
Watches established.
[ls.io-init] done.
Invalid version format (non-numeric data) at /usr/bin/ddclient line 24.
Invalid version format (non-numeric data) at /usr/bin/ddclient line 24.
Invalid version format (non-numeric data) at /usr/bin/ddclient line 24.
Invalid version format (non-numeric data) at /usr/bin/ddclient line 24.
Invalid version format (non-numeric data) at /usr/bin/ddclient line 24.
Invalid version format (non-numeric data) at /usr/bin/ddclient line 24.
Invalid version format (non-numeric data) at /usr/bin/ddclient line 24.
Invalid version format (non-numeric data) at /usr/bin/ddclient line 24.
Invalid version format (non-numeric data) at /usr/bin/ddclient line 24.
<message repeats every few seconds>
```

**Comment by j0nnymoe** [comment]
Looks like there are breaking changes in their newest release, roll back your version.

**Comment by josco95** [comment]
> Looks like there are breaking changes in their newest release, roll back your version.

Can confirm falling back to 3.10.0 stops the scrolling errors and successfully updates my configured dynamic host records, but produces additional errors of their own such as:
Use of uninitialized value $h in hash element at /usr/bin/ddclient line 4101.
Use of uninitialized value $h in hash element at /usr/bin/ddclient line 4103.
Use of uninitialized value $h in hash element at /usr/bin/ddclient line 4104.
Use of uninitialized value $_[0] in sprintf at /usr/bin/ddclient line 2160.

and 

Use of uninitialized value in string ne at /usr/bin/ddclient line 1154.

**Comment by spencerhughes** [comment]
Manually replacing `@PACKAGE_VERSION@` with `v3.11.0` on line 24 of `/usr/bin/ddclient` in the container resolved the issue for me. [The issue has also been reported upstream.](https://github.com/ddclient/ddclient/issues/579)

**Comment by pCoder3000** [comment]
How long to add this fix to master and to docker up? I have the same issue

**Comment by hubertbanas** [comment]
Here is a temporary workaround for anyone interested. This can be dropped to
`custom-cont-init.d/ddclientVersionFix.sh`

```bash
#!/bin/bash

scriptName="$(basename $0)"
ddclientVersion="v3.11.0"
ddclientFile="/usr/bin/ddclient"

echo "${scriptName} - 'ddclientVersion: '${ddclientVersion}'"
echo "${scriptName} - 'ddclientFile: '${ddclientFile}'"

sed -i "s/@PACKAGE_VERSION@/${ddclientVersion}/g" ${ddclientFile}
```

Don't forget to remove it once permanent solution is implemented.

**Comment by saschabrockel** [comment]
Same here.

**Comment by LenardHess** [comment]
I just released v3.11.1 - with the `@PACKAGE_VERSION@` placeholder replaced with the version.
That should mitigate this issue.

**Comment by josco95** [comment]
> I just released v3.11.1 - with the `@PACKAGE_VERSION@` placeholder replaced with the version. That should mitigate this issue.

Confirming my issues as resolved with the release of v3.11.1 - my logs show no errors at all, scrolling or otherwise!
